dome9CloudBots/bots/sql_enable_azure_ad_authentication.py

```diff
@@ -0,0 +1,43 @@
+# What it does: Sets an Azure SQL Server to use Azure AD authentication for Administrators
+# Usage: sql_enable_azure_ad_authentication <azure-ad-admin-email> <azure-ad-admin-sid> <azure-ad-tenant-id>
+# Usage: Example sql_enable_azure_ad_authentication sqladmin@mytenant.onmicrosoft.com 2be17144-2741-1111-ce5e-614a7bb5a9b5
+# Limitations: None
+
+import logging
+from azure.common.credentials import ServicePrincipalCredentials
+from msrestazure.azure_exceptions import CloudError
+from azure.mgmt.sql import SqlManagementClient
+from azure.mgmt.sql.models import ServerAzureADAdministrator
+
+def raise_credentials_error():
+    msg = 'Error! Subscription id or credentials are missing.'
+    logging.info(f'{__file__} - {msg}')
+    return msg
+
+def run_action(credentials, rule, entity, params):
+    logging.info(f'{__file__} - ${run_action.__name__} started')
+    azure_ad_admin_name, azure_ad_user_sid, tenant = params
+    entity_id = entity.get('id')
+    split_id = entity_id.split('/')

```

Comment:
Maybe there is more straight forward way to get those ids and names then splitting the entity's id? In case of change the run will fail.

Reply:
Cleaned this up, removed split and used get() instead